File: NO0001-0100/0001twoSum/twoSum.go
package twosum

//1. 两数之和
//给定一个整数数组 nums 和一个目标值 target，请你在该数组中找出和为目标值的那 两个 整数，并返回他们的数组下标。
//你可以假设每种输入只会对应一个答案。但是，你不能重复利用这个数组中同样的元素。
//示例:
//	给定 nums = [2, 7, 11, 15], target = 9
//	因为 nums[0] + nums[1] = 2 + 7 = 9
//	所以返回 [0, 1]
//来源：力扣（LeetCode）
//链接：https://leetcode-cn.com/problems/two-sum
//著作权归领扣网络所有。商业转载请联系官方授权，非商业转载请注明出处。

func TwoSum(nums []int, target int) []int {
	for i, num1 := range nums[:len(nums)-1] {
		for j, num2 := range nums[i+1:] {
			if num1+num2 == target {
				return []int{i, i + 1 + j}
			}
		}
	}
	return []int{}
}

//space: O(1)
//time : O(n^2)

File: NO0001-0100/0001twoSum/twoSum.py
#coding=utf-8
import unittest

class TestCaseMySolution(unittest.TestCase):
    def testCase01(self):
        nums = [2,7,11,15]
        target= 9
        want = [0,1]

        got = Solution().twoSum(nums, target)
        self.assertEqual(want,got)

    def testCase02(self):
        nums = [3,2,4]
        target= 6
        want = [1,2]

        got = Solution().twoSum(nums, target)
        self.assertEqual(want,got)

        # self.assertEqualTure('FOO'.isupper())
        # self.assertFalse('Foo'.isupper())
        # self.assertEqual('foo',upper(),'FOO')
        # self.assertEqual('hello world'.split(),['hello','world']) with self.assertEqualRaise(TypeError):

class Solution:
    def twoSum(self, nums, target) :
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        for i in range(0,len(nums)-1) :
            for j in range(i+1,len(nums)) :
                if nums[i]+nums[j] == target :
                    return [i,j]
        return []


if __name__ == '__main__':
    suite = unittest.TestSuite()
    suite.addTest(unittest.TestLoader().loadTestsFromTestCase(TestCaseMySolution))
    testResult=unittest.TextTestRunner(verbosity=2).run(suite)

    print("testsRun:%s" % testResult.testsRun)
    print("failures:%s" % len(testResult.failures))
    print("errors  :%s" % len(testResult.errors))
    print("skipped :%s" % len(testResult.skipped))
